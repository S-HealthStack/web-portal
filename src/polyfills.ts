import smoothScroll from 'smoothscroll-polyfill';
import _isNull from 'lodash/isNull';
import _isObject from 'lodash/isObject';
import browser from 'src/common/utils/browser';

type ArgumentTypes<F extends () => unknown> = F extends (...args: infer A) => unknown ? A : never;

// safari smooth scroll fix
if (browser.isSafari) {
  smoothScroll.polyfill();

  const prevScrollTo = window.scrollTo;

  // issue: https://github.com/iamdustan/smoothscroll/issues/161
  Object.assign(window, {
    scrollTo: ((...args) => {
      if (_isObject(args[0])) {
        args[0].left = _isNull(args[0].left) ? undefined : args[0].left;
        args[0].top = _isNull(args[0].top) ? undefined : args[0].top;
      } else {
        args[0] = _isNull(args[0]) ? undefined : args[0];
        args[1] = _isNull(args[1]) ? undefined : args[1];
      }

      prevScrollTo(...(args as ArgumentTypes<typeof window.scrollTo>));
    }) as typeof window.scrollTo,
  });
}
