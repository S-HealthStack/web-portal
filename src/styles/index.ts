export { default as theme } from './theme';
export {
  colorsFn as colors,
  typographyFn as typography,
  boxShadowFn as boxShadow,
  animationFn as animation,
} from './themeHelpers';
export { default as GlobalStyles } from './GlobalStyles';
export { px } from './utils';
