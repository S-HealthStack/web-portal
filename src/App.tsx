import React from 'react';
import { Provider } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { ConnectedRouter } from 'connected-react-router';
import { ThemeProvider } from 'styled-components';
import { DndProvider } from 'react-dnd';
import { TouchBackend } from 'react-dnd-touch-backend';

import { store } from 'src/modules/store/store';
import { history, Path } from 'src/modules/navigation/store';
import PrivateRoute from 'src/modules/navigation/private-route/PrivateRoute';
import MainLayout from 'src/modules/main-layout/MainLayout';
import SignInScreen from 'src/modules/auth/signin/SignInScreen';
import AccountActivationScreen from 'src/modules/auth/signin/AccountActivationScreen';
import CreateStudyScreen from 'src/modules/auth/signin/CreateStudyScreen';
import { TooltipProvider, TooltipsList } from 'src/common/components/Tooltip';
import { theme, GlobalStyles } from 'src/styles';

import 'src/modules/auth/authProvider';

const App = () => (
  <ThemeProvider theme={theme}>
    <TooltipProvider>
      <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
        <Provider store={store}>
          <GlobalStyles />
          <ConnectedRouter history={history}>
            <Switch>
              <Route path={Path.SignIn} component={SignInScreen} />
              <Route path={Path.AccountActivation} component={AccountActivationScreen} />
              <PrivateRoute path={Path.CreateStudy}>
                <CreateStudyScreen />
              </PrivateRoute>
              <PrivateRoute>
                <MainLayout />
              </PrivateRoute>
            </Switch>
          </ConnectedRouter>
        </Provider>
        <TooltipsList />
      </DndProvider>
    </TooltipProvider>
  </ThemeProvider>
);

export default App;
