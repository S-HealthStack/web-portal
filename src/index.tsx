import React from 'react';
import { createRoot } from 'react-dom/client';

import printAppVersion from 'src/common/utils/printAppVersion';

import './polyfills';
import App from './App';

printAppVersion();

const container = document.getElementById('root');
const root = createRoot(container as HTMLElement);

root.render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
