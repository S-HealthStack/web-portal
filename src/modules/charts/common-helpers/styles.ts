import styled, { css } from 'styled-components';

import { colors, typography, px, animation } from 'src/styles';
import { LINE_CLASS_NAME } from '../common-components/Line';
import { AREA_CLASS_NAME } from '../common-components/Area';
import { DOT_CLASS_NAME } from '../common-components/Dot';

interface ContainerProps {
  width: number;
  height: number;
}

export const DOTS_CONTAINER_CLASS_NAME = 'dots';
export const X_AXIS_CLASS_NAME = 'xAxis';
export const Y_AXIS_CLASS_NAME = 'yAxis';
export const ZOOM_CLASS_NAME = 'zoom';

export const Container = styled.div.attrs<ContainerProps>(({ width, height }) => ({
  style: {
    width: px(width),
    height: px(height),
    position: 'relative',
  },
}))<ContainerProps>``;

export const TooltipContent = styled.div`
  display: flex;
  align-items: center;
`;

export const TooltipNameBlock = styled.div`
  display: flex;
  margin-right: ${px(8)};

  p {
    ${typography.labelSemibold};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipDataBlock = styled.div`
  display: flex;
  flex-direction: column;

  p {
    ${typography.labelRegular};
    color: ${colors.surface};
    margin: 0;
  }
`;

export const TooltipColorPoint = styled.div<{ color: string }>`
  width: ${px(12)};
  height: ${px(12)};
  background-color: ${({ color }) => color};
  border-radius: 50%;
  margin-right: ${px(8)};
`;

export const brushOverlayStyles = css`
  fill: ${colors.primary};
  fill-opacity: 0.1;
  pointer-events: none;
`;

export const brushSelectionStyles = css`
  fill: ${colors.primary};
  fill-opacity: 1;
`;

export const brushHandleCommonStyles = css`
  stroke-width: ${px(1)};
  stroke: ${colors.primary};
  stroke-opacity: 0.3;
  fill: ${colors.onPrimary};
`;

export const StyledSvg = styled.svg<{
  focusWidth: number;
  $contextVisible: boolean;
}>`
  .${X_AXIS_CLASS_NAME} {
    .tick line {
      opacity: 0;
    }

    .tick {
      :first-child text {
        text-anchor: start;
      }
      :last-child text {
        text-anchor: end;
      }
    }
  }

  .${Y_AXIS_CLASS_NAME} {
    .tick line {
      stroke: ${colors.updBackgoundLight};
    }
  }

  .context {
    opacity: ${({ $contextVisible }) => ($contextVisible ? 1 : 0)};
    transition: opacity 0.3s ${animation.defaultTiming};
    .brushX {
      .overlay {
        ${brushOverlayStyles};
        height: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        height: ${px(22)};
        fill: url(#horizontalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(22)};
        width: ${px(6)};
      }
    }
    .brushY {
      .overlay {
        ${brushOverlayStyles};
        width: ${px(4)};
      }
      .selection {
        ${brushSelectionStyles};
        width: ${px(22)};
        fill: url(#verticalGradient);
      }
      .handle {
        ${brushHandleCommonStyles};
        height: ${px(6)};
        width: ${px(22)};
      }
    }
  }

  .focus {
    cursor: move;
  }

  .${LINE_CLASS_NAME} {
    clip-path: url(#clip);
    transition: stroke-opacity 0.3s ${animation.defaultTiming};
  }

  .${AREA_CLASS_NAME} {
    clip-path: url(#clip);
    transition: fill-opacity 0.3s ${animation.defaultTiming};
  }

  .${DOTS_CONTAINER_CLASS_NAME} {
    clip-path: url(#clip);
    .${DOT_CLASS_NAME} {
      transition: fill-opacity 0.3s ${animation.defaultTiming};
    }
  }

  .zoom {
    fill: none;
    pointer-events: all;
  }
`;
