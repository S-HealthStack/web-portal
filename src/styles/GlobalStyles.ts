import { createGlobalStyle } from 'styled-components';
import { px } from 'src/styles/utils';

export const previewDropdownMenuClassName = 'preview-dropdown-menu';
export const PREVIEW_SCALE = 0.75;

export default createGlobalStyle`
  * {
    box-sizing: border-box;
  }

  html {
    font-size: 100%;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  html, body {
    height: 100%;
    width: 100%;
    margin: 0;
    position: fixed;
    overflow: hidden;
    overscroll-behavior-y: none;
    background: ${({ theme }) => theme.colors.background};
    user-select: none;
    -webkit-user-drag: none;
    font-family: Inter, sans-serif;
  }
  
  input,
  input:focus {
    ::-webkit-textfield-decoration-container {
      visibility: hidden;
    }
  }

  #root {
    height: 100%;
    width: 100%;
    display: flex;
    flex-direction: column;
    overflow: hidden;
  }
  
  *:focus,
  *:focus-visible {
    outline: none;
  }
  
  .recharts-surface {
    overflow: visible;
  }

  .${previewDropdownMenuClassName} {
    border: none;
    filter: drop-shadow(
      0 
      ${px(5 * PREVIEW_SCALE)} 
      ${px(12 * PREVIEW_SCALE)} 
      rgba(0, 0, 0, 0.12)
    ); // TODO unknown color
    margin-top: ${px(8 * PREVIEW_SCALE)};
  }
`;
