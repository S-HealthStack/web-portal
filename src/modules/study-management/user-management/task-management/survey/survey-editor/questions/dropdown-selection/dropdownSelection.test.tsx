import React from 'react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { act, render, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from 'styled-components/';
import { TouchBackend } from 'react-dnd-touch-backend';
import { DndProvider } from 'react-dnd';
import { theme } from 'src/styles';
import handler from './index';

/* eslint-disable @typescript-eslint/no-explicit-any */

describe('dropdown-selection', () => {
  it('type should be dropdown', async () => {
    expect(handler.type).toEqual('dropdown');
  });

  const emptyQuestion = {
    type: 'dropdown',
    id: 'survey_question1',
    title: '',
    description: '',
    answers: [
      { id: 'survey_question2', value: 'Enter option 1' },
      { id: 'survey_question3', value: 'Enter option 2' },
    ],
    options: {
      optional: false,
      includeOther: false,
    },
  };

  it('should return empty question', async () => {
    expect(handler.createEmpty()).toEqual(emptyQuestion);
  });

  it('isEmpty should return true', async () => {
    expect(handler.isEmpty(emptyQuestion as any)).toBeTrue();
  });

  it('[NEGATIVE] isEmpty should return true if question is undefined', async () => {
    expect(handler.isEmpty(undefined as any)).toBeTrue();
  });

  it('[NEGATIVE] isEmpty should return true if question answers is undefined', async () => {
    expect(handler.isEmpty({ ...emptyQuestion, answers: undefined } as any)).toBeTrue();
  });

  const notEmptyQuestionItem = {
    id: '1',
    type: 'dropdown',
    title: 'Title',
    description: 'Description',
    answers: [
      { id: 'survey_question2', value: 'Option 1' },
      { id: 'survey_question3', value: 'Option 2' },
    ],
    options: {
      optional: false,
      includeOther: false,
    },
  };

  it('isEmpty should return false', async () => {
    expect(handler.isEmpty(notEmptyQuestionItem as any)).toBeFalse();
  });

  it('convertFromOtherType should return converted question with similar answers', async () => {
    const qi = {
      type: 'single',
      id: 'Question1',
      title: 'Title',
      description: 'Description',
      skipLogic: undefined,
      options: {
        optional: false,
        includeOther: false,
      },
      answers: [{ answer1: 'answer1' }, { answer2: 'answer2' }],
    };
    expect(handler.convertFromOtherType(qi as any)).toEqual({
      ...qi,
      type: 'dropdown',
      options: {
        optional: false,
        includeOther: false,
      },
      answers: [{ answer1: 'answer1' }, { answer2: 'answer2' }],
    });
  });

  it('convertFromOtherType should return converted question with empty answers', async () => {
    const qi = {
      type: 'images',
      id: 'Question1',
      title: 'Title',
      description: 'Description',
      skipLogic: undefined,
      options: {
        optional: false,
        includeOther: false,
      },
      answers: [{ answer1: 'answer1' }, { answer2: 'answer2' }],
    };
    expect(handler.convertFromOtherType(qi as any)).toEqual({
      ...qi,
      type: 'dropdown',
      options: {
        optional: false,
        includeOther: false,
      },
      answers: [
        { id: 'survey_question2', value: 'Enter option 1' },
        { id: 'survey_question3', value: 'Enter option 2' },
      ],
    });
  });

  it('[NEGATIVE] convertFromOtherType should return empty question', async () => {
    expect(handler.convertFromOtherType({} as any)).toEqual({
      type: 'dropdown',
      id: undefined,
      title: undefined,
      description: undefined,
      answers: [
        { id: 'survey_question2', value: 'Enter option 1' },
        { id: 'survey_question3', value: 'Enter option 2' },
      ],
      options: {
        optional: false,
        includeOther: false,
      },
      skipLogic: undefined,
    });
  });

  const taskItemFromApi = {
    name: 'name',
    sequence: 1,
    type: 'QUESTION',
    contents: {
      title: 'Question Title',
      explanation: 'Question Description',
      required: true,
      type: 'CHOICE',
      properties: {
        options: [
          { id: 'survey_question2', value: 'Option 1' },
          { id: 'survey_question3', value: 'Option 2' },
        ],
      },
    },
  };

  it('fromApi should return converted question', async () => {
    expect(handler.fromApi(taskItemFromApi as any)).toEqual({
      type: 'dropdown',
      id: 'name',
      title: 'Question Title',
      description: 'Question Description',
      answers: [
        {
          id: 'survey_question1',
          value: 'Option 1',
        },
        {
          id: 'survey_question2',
          value: 'Option 2',
        },
      ],
      options: {
        optional: false,
        includeOther: false,
      },
    });
  });

  it('[NEGATIVE] fromApi should return undefined if task type is not a question', async () => {
    expect(handler.fromApi({ ...taskItemFromApi, type: 'ROW' })).toBeUndefined();
  });

  it('[NEGATIVE] fromApi should return undefined if question type is not an dropdown', async () => {
    expect(
      handler.fromApi({
        ...taskItemFromApi,
        contents: { ...taskItemFromApi.contents, type: 'TEXT' },
      } as any)
    ).toBeUndefined();
  });

  it('toApi should return converted question', async () => {
    expect(handler.toApi(notEmptyQuestionItem as any)).toEqual({
      name: '1',
      type: 'QUESTION',
      sequence: 0,
      contents: {
        title: 'Title',
        explanation: 'Description',
        required: true,
        type: 'CHOICE',
        properties: {
          tag: 'DROPDOWN',
          options: [{ value: 'Option 1' }, { value: 'Option 2' }],
        },
      },
    });
  });

  it('[NEGATIVE] toApi should not crash with empty data', async () => {
    expect(handler.toApi({} as any)).toEqual({
      name: undefined,
      type: 'QUESTION',
      sequence: 0,
      contents: {
        title: undefined,
        explanation: undefined,
        required: false,
        type: 'CHOICE',
        properties: {
          tag: 'DROPDOWN',
          options: [],
        },
      },
    });
  });

  it('transformAnswersOnQuestionChange should return empty object', async () => {
    expect(
      handler.transformAnswersOnQuestionChange({
        question: notEmptyQuestionItem as any,
        previousQuestion: { ...notEmptyQuestionItem, type: 'image' } as any,
        answers: { survey_question2: 1 },
      })
    ).toEqual({});
  });

  it('transformAnswersOnQuestionChange should filter invalid answers', async () => {
    expect(
      handler.transformAnswersOnQuestionChange({
        question: notEmptyQuestionItem as any,
        previousQuestion: notEmptyQuestionItem as any,
        answers: { survey_question2: 1, other: 1 },
      })
    ).toEqual({ survey_question2: 1 });
  });

  it('[NEGATIVE] transformAnswersOnQuestionChange should return empty object if all answers are invalid', async () => {
    expect(
      handler.transformAnswersOnQuestionChange({
        question: notEmptyQuestionItem as any,
        previousQuestion: {
          ...notEmptyQuestionItem,
          answers: [{ id: '1', value: '1' }],
        } as any,
        answers: undefined as any,
      })
    ).toEqual({});
  });

  it('isPreviewQuestionAnswered should return true', async () => {
    expect(
      handler.isPreviewQuestionAnswered({
        question: notEmptyQuestionItem as any,
        answers: { survey_question2: 1 },
      })
    ).toBeTrue();
  });

  it('[NEGATIVE] isPreviewQuestionAnswered should return true if question is not defined but answers are defined', async () => {
    expect(
      handler.isPreviewQuestionAnswered({
        question: undefined as any,
        answers: { survey_question2: 1 },
      })
    ).toBeTrue();
  });

  it('[NEGATIVE] isPreviewQuestionAnswered should return false if answers are not defined', async () => {
    expect(
      handler.isPreviewQuestionAnswered({
        question: notEmptyQuestionItem as any,
        answers: undefined as any,
      })
    ).toBeFalse();
  });

  it('[NEGATIVE] isPreviewQuestionAnswered should return false if question is not defined and answers are not defined', async () => {
    expect(
      handler.isPreviewQuestionAnswered({
        question: undefined as any,
        answers: undefined as any,
      })
    ).toBeFalse();
  });

  it('renderEditorContent should render editor content', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    const { baseElement, getByTestId, queryAllByTestId } = render(
      <ThemeProvider theme={theme}>
        <DndProvider backend={TouchBackend} options={{ enableMouseEvents: true }}>
          {handler.renderEditorContent({
            surveyId: '',
            question: notEmptyQuestionItem as any,
            onChange,
            confirmOptionRemoval,
          })}
        </DndProvider>
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();

    const addOptionField = getByTestId('question-card-add-option');
    const newOptionTitle = 'New Option';

    await act(async () => userEvent.type(addOptionField, newOptionTitle));
    expect(addOptionField).toHaveValue(newOptionTitle);
    await waitFor(
      () => {
        expect(onChange).toHaveBeenCalled();
      },
      // required because survey input has a debounce for invoking onChange callback
      { timeout: 2500 }
    );

    const firstOptionDeleteButton = queryAllByTestId('delete-option-button')[1];
    await act(async () => userEvent.click(firstOptionDeleteButton));
    await waitFor(() => {
      expect(onChange).toHaveBeenCalled();
    });
  });

  it('[NEGATIVE] renderEditorContent should return null if question is undefined', async () => {
    const onChange = jest.fn();
    const confirmOptionRemoval = jest.fn();

    expect(
      handler.renderEditorContent({
        surveyId: '',
        question: undefined as any,
        onChange,
        confirmOptionRemoval,
      })
    ).toBeNull();
  });

  it('renderPreviewContent should render preview content', async () => {
    const onAnswersChange = jest.fn();
    const onChange = jest.fn();

    const { baseElement } = render(
      <ThemeProvider theme={theme}>
        {handler.renderPreviewContent({
          question: notEmptyQuestionItem as any,
          answers: { survey_question2: 1 },
          onAnswersChange,
          onChange,
        })}
      </ThemeProvider>
    );

    expect(baseElement).toMatchSnapshot();
  });

  it('[NEGATIVE] renderPreviewContent should return null if question is undefined', async () => {
    const onAnswersChange = jest.fn();
    const onChange = jest.fn();

    expect(
      handler.renderPreviewContent({
        question: undefined as any,
        answers: { survey_question2: 1 },
        onChange,
        onAnswersChange,
      })
    ).toBeNull();
  });

  it('[NEGATIVE] renderPreviewContent should return null if answers is undefined', async () => {
    const onAnswersChange = jest.fn();
    const onChange = jest.fn();

    expect(
      handler.renderPreviewContent({
        question: notEmptyQuestionItem as any,
        answers: undefined as any,
        onChange,
        onAnswersChange,
      })
    ).toBeNull();
  });
});
