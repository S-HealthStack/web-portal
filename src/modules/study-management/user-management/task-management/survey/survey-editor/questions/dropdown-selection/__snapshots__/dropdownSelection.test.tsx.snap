// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dropdown-selection renderEditorContent should render editor content 1`] = `
.c12 {
  position: absolute;
  will-change: opacity,transform;
  border-radius: 50%;
  -webkit-animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  animation: bykygw 600ms cubic-bezier(0.4,0,0.2,1) forwards;
  z-index: 0;
  background-color: #E3EAFB;
  pointer-events: none;
}

.c9 {
  position: relative;
  z-index: 0;
  display: -webkit-box !important;
  display: -webkit-flex !important;
  display: -ms-flexbox !important;
  display: flex !important;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border-radius: 0.25rem;
  width: 100%;
  white-space: nowrap;
  padding: 0;
  overflow: hidden;
  -webkit-transition-duration: 0;
  transition-duration: 0;
  -webkit-transition-property: background-color;
  transition-property: background-color;
  height: 1.5rem;
  width: 1.5rem;
  border: none;
  background-color: transparent;
}

.c9:hover {
  cursor: pointer;
}

.c9 > div {
  color: #4475E3;
}

.c9 > div svg {
  fill: #4475E3;
}

.c9:hover:enabled > div {
  color: #2F65E0;
}

.c9:hover:enabled > div svg {
  fill: #2F65E0;
}

.c9:hover:enabled:active > div {
  color: #2057D5;
}

.c9:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c9:disabled > div {
  color: #B3C6F1;
}

.c9:disabled > div svg {
  fill: #B3C6F1;
}

.c9:focus-visible {
  outline: 0.125rem solid #C7D6F7;
}

.c9:focus-visible > div {
  color: #4475E3;
}

.c9:focus-visible > div svg {
  fill: #4475E3;
}

.c9:hover:enabled {
  background-color: #ECF1FC;
}

.c9:hover:enabled:active > div {
  color: #2057D5;
}

.c9:hover:enabled:active > div svg {
  fill: #2057D5;
}

.c9:focus-visible:enabled {
  background-color: #C7D6F7;
  outline: none;
}

.c11 {
  position: relative;
  z-index: 2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: translate3d(0,0,0.0625rem);
  -ms-transform: translate3d(0,0,0.0625rem);
  transform: translate3d(0,0,0.0625rem);
}

.c4 {
  -webkit-scrollbar-width: thin;
  -moz-scrollbar-width: thin;
  -ms-scrollbar-width: thin;
  scrollbar-width: thin;
  -webkit-scrollbar-color: #C7D6F7 #FFFFFF;
  -moz-scrollbar-color: #C7D6F7 #FFFFFF;
  -ms-scrollbar-color: #C7D6F7 #FFFFFF;
  scrollbar-color: #C7D6F7 #FFFFFF;
}

.c4::-webkit-scrollbar {
  width: 1.25rem;
  height: 1.25rem;
}

.c4::-webkit-scrollbar-track {
  background: #FFFFFF;
  border-radius: 0.25rem;
  margin: 0.0625rem 0;
}

.c4::-webkit-scrollbar-thumb {
  background-color: #C7D6F7;
  border-radius: 0.625rem;
  border: 0.5rem solid #FFFFFF;
  width: 0.25rem;
  min-height: 1.25rem;
}

.c5 {
  resize: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  box-sizing: border-box;
  outline: none;
  width: 100%;
  padding: 0;
  margin: 0;
  border: none;
  background: transparent;
  overflow-y: hidden;
  caret-color: #4475E3;
  overflow-x: hidden;
  resize: none;
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #808080;
  min-height: 1.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  position: relative;
}

.c3 {
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  margin: 0.5625rem 0 0.4375rem !important;
  color: #474747;
}

.c6::-webkit-input-placeholder {
  color: #A1A1A1;
}

.c6::-moz-placeholder {
  color: #A1A1A1;
}

.c6:-ms-input-placeholder {
  color: #A1A1A1;
}

.c6::placeholder {
  color: #A1A1A1;
}

.c8 {
  height: 2.5rem;
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.5rem;
  grid-template-columns: repeat(2,1.5rem);
}

.c10 {
  visibility: hidden;
}

.c13 {
  cursor: move !important;
}

.c1 {
  min-height: 2.5rem;
  height: 2.5rem;
  border-radius: 0.25rem;
  display: grid;
  margin: 0.25rem 0;
  gap: 0.25rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  background-color: #FFFFFF;
}

.c2 {
  grid-template-columns: 2.5rem 1fr 4.5rem;
  opacity: 1;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
}

.c2 .c7 {
  opacity: 0;
  -webkit-transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
  transition: opacity 300ms cubic-bezier(0.45,0.05,0,1);
}

.c2:hover {
  background-color: #F7F8FA;
}

.c2:hover .c7 {
  opacity: 1;
}

.c14 {
  grid-template-columns: 2.5rem 1fr;
}

<body>
  <div>
    <div
      class="c0"
    >
      <div
        class="c1 c2"
      >
        <div
          class="c3"
        >
          <svg />
        </div>
        <textarea
          class=" c4 c5 c6"
          maxlength="256"
          placeholder="Enter option"
          rows="1"
          style="overflow-y: hidden; height: 0rem;"
        >
          Option 1
        </textarea>
        <div
          class="c7 c8"
        >
          <button
            aria-label="Delete Option"
            class="c9 c10"
            data-testid="delete-option-button"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c11"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c12"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
          <button
            aria-label="Drag and Drop Question"
            class="c9 c13"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c11"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c12"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
        </div>
      </div>
      <div
        class="c1 c2"
      >
        <div
          class="c3"
        >
          <svg />
        </div>
        <textarea
          class=" c4 c5 c6"
          maxlength="256"
          placeholder="Enter option"
          rows="1"
          style="overflow-y: hidden; height: 0rem;"
        >
          Option 2
        </textarea>
        <div
          class="c7 c8"
        >
          <button
            aria-label="Delete Option"
            class="c9 c10"
            data-testid="delete-option-button"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c11"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c12"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
          <button
            aria-label="Drag and Drop Question"
            class="c9 c13"
            fill="text"
            style="overflow: hidden; position: relative;"
          >
            <div
              class="c11"
              data-testid="content"
            >
              <svg />
            </div>
            <div
              class="c12"
              color="primaryLightPressed"
              opacity="1,1"
              scale="0"
              style="display: none; width: 0rem; height: 0rem; top: 0rem; left: 0rem;"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="c1 c14"
    >
      <div
        class="c3"
      >
        <svg />
      </div>
      <textarea
        class=" c4 c5 c6"
        data-testid="question-card-add-option"
        placeholder="Add option"
        rows="1"
        style="overflow-y: hidden; height: 0rem;"
      />
    </div>
  </div>
</body>
`;

exports[`dropdown-selection renderPreviewContent should render preview content 1`] = `
.c9 {
  display: inline-grid;
  gap: 0.25rem;
  grid-template-columns: 2.5rem;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c11 {
  width: 2.5rem;
  height: 2.5rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c7 {
  display: inline-block;
}

.c7 input {
  display: none;
}

.c7 .c10 > svg > :nth-child(1) {
  -webkit-transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  fill: #4475E3;
}

.c7 .c10 > svg > :nth-child(2) {
  -webkit-transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
  transition: stroke-color 300ms cubic-bezier(0.45,0.05,0,1), fill 300ms cubic-bezier(0.45,0.05,0,1);
}

.c7:hover > .c8 {
  cursor: pointer;
}

.c0 {
  position: relative;
  width: 100%;
}

.c6 {
  width: 2.5rem;
  height: 2.5rem;
  grid-area: left-icon;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c13 {
  font-family: Inter,sans-serif;
  font-size: 0.875rem;
  line-height: 1.3125rem;
  font-weight: 400;
  font-style: normal;
  color: #474747;
  grid-area: label;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.c14 {
  grid-area: right-icon;
}

.c3 {
  width: 100%;
  height: 2.625rem;
  padding: 0 1rem;
  padding-right: 0.5rem;
  box-sizing: border-box;
  display: grid;
  grid-template-areas: 'left-icon label right-icon';
  grid-template-columns: 2.5rem 1fr 24px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #FFFFFF;
  border: 0.0625rem solid transparent;
  border-top-width: 0;
  -webkit-transition: all 300ms cubic-bezier(0.45,0.05,0,1);
  transition: all 300ms cubic-bezier(0.45,0.05,0,1);
  padding-left: 0.5rem;
}

.c3:hover {
  cursor: pointer;
}

.c4 {
  border-top-width: 0.0625rem;
  border-color: !important;
  border-radius: 0.25rem;
}

.c4:hover {
  border-color: #2F65E0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 0.4375rem solid transparent;
  border-top: 0.4375rem solid transparent;
  border-radius: 0.25rem;
  outline: #B3C6F1 solid 0.0625rem;
  outline-color: #4475E3;
  width: 19.5rem;
  min-width: 19.5rem;
}

.c1:hover {
  outline: #4475E3 solid 0.0625rem;
}

.c1 .c5 {
  display: none;
}

.c1 .c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 1.125rem;
  border-color: transparent !important;
}

.c1 .c2:hover {
  border-color: transparent !important;
}

.c1 .c12 {
  font-size: 1rem;
}

<body>
  <div>
    <div
      class="c0 c1"
    >
      <div
        class="c2 c3 c4"
        data-testid="value-item"
      >
        <div
          class="c5 c6"
        >
          <div
            class="c7 "
            kind="radio"
          >
            <label
              class="c8 c9"
              for="1"
              kind="radio"
            >
              <div
                class="c10 c11"
              >
                <svg />
              </div>
            </label>
            <input
              checked=""
              id="1"
              type="checkbox"
            />
          </div>
        </div>
        <div
          class="c12 c13"
        >
          Option 1
        </div>
        <svg
          class="c14"
        />
      </div>
    </div>
  </div>
</body>
`;
